client/src/components/MessageListPanel/MessageListPanel.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Message from '../Message';
+import './message-list-panel.css';
+
+const MessageListPanel = ({ messages }) => {
+    const messagesComponenets = messages.map(
+        (message, index) => (<Message key={index} message={message} />)
+    );
+
+    return (
+        <div className="message-list-panel">
+            {messagesComponenets}
+        </div>
+    )
+};
+
+MessageListPanel.propTypes = {
+    messages: PropTypes.arrayOf(PropTypes.shape({
+        data: PropTypes.string,
+        from: PropTypes.shape({
```

Comment:
Is this required here ?

Reply:
care to explain? 
do you mean do I need to write it again event if i wrote it in Message.jsx?
I just like to be explicit and not use Object or just array

Reply:
It looks like you understood what I meant. I don't think MessageListPanel needs to know the structure of my Message and just use the object so in future if I want to change the object I would need to change it only in one place

Reply:
fair enough